### PR TITLE
feat: public params artifact

### PR DESCRIPTION
**Pull request by Autoparallel**
WIP.
- [x] Works locally!
- [x] This won't work until we make web-prover public or get a key to be able to pull that in as a crate.
- [x] Test generated `AuxParams` in `web-prover`

---

Waiting on https://github.com/pluto/web-prover/pull/331

### CODE REVIEW

**Comment by Autoparallel** [comment]
Added a bit more here: serialize the `ProverKey` and `VerifierKey`

But won't work until changes are made in `client-side-prover`.

**Comment by mattes** [comment]
Github CI working ([as seen here](https://github.com/pluto/web-prover-circuits/actions/runs/11963222945/job/33353260197)), but failing for other reasons here. Should just work in main.

**Comment by Autoparallel** [comment]
Not sure why I had to deal with that conflict when I pushed up a change right before it. Sometimes I'm so dumb with git I swear

**Comment by Autoparallel** [comment]
Want to merge in https://github.com/pluto/web-prover/pull/331 prior to merging this and then pin to the correct commit for `proofs` crate.

**Review by 0xJepsen — APPROVED** [review]
Nice this looks great! How big are these setup artifacts?

**Comment by Autoparallel** [comment]
You can pull them up yourself if you go to the action

File: Makefile [hunk 1]
```diff
@@ -4,12 +4,15 @@ TARGET_DIRS := $(wildcard builds/target_*b)
...
+# Extract target sizes (e.g., "512b", "1024b") from directory names
+TARGET_SIZES := $(patsubst builds/target_%,%,$(TARGET_DIRS))
+
 # Create artifacts directories
 $(shell mkdir -p $(addsuffix /artifacts,$(TARGET_DIRS)))
 
 # Default target
 .PHONY: all clean
-all: build
+all: buildmak
```
Comment (lonerapier) [inline Makefile:15]: is this a typo? can't run this in local

File: Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+[package]
+name = "create-pp"
```
Comment (lonerapier) [inline Cargo.toml:2]: minor nit but package name needs some love =D

**Review by lonerapier — APPROVED** [review]
this going to make our life so much easier

File: Makefile (hunk 1 above)
Comment (Autoparallel) [reply Makefile:15]: yes a typo oops

File: Cargo.toml (hunk 2 above)
Comment (Autoparallel) [reply Cargo.toml:2]: lol what do you suggest? Go ahead and edit it if you want!

**Review by 0xJepsen — APPROVED** [review]
nice!

**Pull request merged**